Dockerfile

```diff
@@ -10,6 +11,26 @@ RUN ./node_modules/.bin/gulp build
 RUN npm run lint:js -- --max-warnings=0
 RUN npm run lint:css
 
+FROM debian as react-builder
+
+# Install Volta, to use the same toolchain (Node+npm) as developers locally:
+RUN apt-get update && apt-get install -y curl ca-certificates --no-install-recommends
+ENV VOLTA_HOME $HOME/.volta
+ENV PATH $VOLTA_HOME/bin:$PATH
+RUN curl https://get.volta.sh | bash
+
+WORKDIR /app
+COPY privaterelay/locales ./privaterelay/locales
+# TODO: Move static files from /static/ to /frontend/public/ and remove this COPY:
+COPY static ./static
+
+WORKDIR /app/frontend
+COPY frontend ./
+RUN npm install
+RUN npm run lint -- --max-warnings=0
+RUN npm run test
```

Comment:
Yes, that makes sense - I might need some help with that though.

As I understand it, in the Dockerfile we currently temporarily load an image with Node available so that we can build the frontend assets. Then the build output is copied over to our actual Docker image, which doesn't include Node and all that. In other words, I don't think we're able to run npm scripts referencing Node modules in the `test` state in CircleCI? Do you know if/how we can run that in the context of that image, or is there another approach you'd suggest?

Reply:
We may want to loop in SRE at this point. @Vinnl is raising a good point. We only had Node in the docker file in the earliest steps. Moving to this front-end will require more dependencies throughout the pipeline.

Reply:
I've seen projects with multiple `Dockerfile`s - e.g., `Dockerfile-frontend`. If we moved this from an intermediate image to its own Dockerfile, maybe we could copy the built files into a shared space in the CI job so it's available when we build the main app container image.

Reply:
OK, figured it out with some help from Aaron. We decided to build the frontend in the regular CircleCI environment rather than attempting to reuse partial Docker images to avoid some complexity, so I used the Circle's Node.js "orb", which is apparently a CircleCI thing for reusable configurations. b1cdf5026e091cd34228e39a1e0ba1fb116cfd7e